presto-hive/src/main/java/com/facebook/presto/hive/util/Statistics.java

```diff
@@ -49,31 +83,186 @@ public static PartitionStatistics merge(PartitionStatistics first, PartitionStat
     }
 
     public static HiveBasicStatistics reduce(HiveBasicStatistics first, HiveBasicStatistics second, ReduceOperator operator)
     {
         return new HiveBasicStatistics(
-                reduce(first.getFileCount(), second.getFileCount(), operator),
-                reduce(first.getRowCount(), second.getRowCount(), operator),
-                reduce(first.getInMemoryDataSizeInBytes(), second.getInMemoryDataSizeInBytes(), operator),
-                reduce(first.getOnDiskDataSizeInBytes(), second.getOnDiskDataSizeInBytes(), operator));
+                reduce(first.getFileCount(), second.getFileCount(), operator, false),
+                reduce(first.getRowCount(), second.getRowCount(), operator, false),
+                reduce(first.getInMemoryDataSizeInBytes(), second.getInMemoryDataSizeInBytes(), operator, false),
+                reduce(first.getOnDiskDataSizeInBytes(), second.getOnDiskDataSizeInBytes(), operator, false));
     }
 
-    private static OptionalLong reduce(OptionalLong first, OptionalLong second, ReduceOperator operator)
+    public static Map<String, HiveColumnStatistics> merge(
+            Map<String, HiveColumnStatistics> first,
+            OptionalLong firstRowCount,
+            Map<String, HiveColumnStatistics> second,
+            OptionalLong secondRowCount)
+    {
+        // only keep columns that have statistics for both sides
+        Set<String> columns = intersection(first.keySet(), second.keySet());
+        return columns.stream()
+                .collect(toImmutableMap(
+                        column -> column,
+                        column -> merge(first.get(column), firstRowCount, second.get(column), secondRowCount)));
+    }
+
+    public static HiveColumnStatistics merge(HiveColumnStatistics first, OptionalLong firstRowCount, HiveColumnStatistics second, OptionalLong secondRowCount)
+    {
+        return new HiveColumnStatistics(
+                mergeIntegerStatistics(first.getIntegerStatistics(), second.getIntegerStatistics()),
+                mergeDoubleStatistics(first.getDoubleStatistics(), second.getDoubleStatistics()),
+                mergeDecimalStatistics(first.getDecimalStatistics(), second.getDecimalStatistics()),
+                mergeDateStatistics(first.getDateStatistics(), second.getDateStatistics()),
+                mergeBooleanStatistics(first.getBooleanStatistics(), second.getBooleanStatistics()),
+                reduce(first.getMaxColumnLength(), second.getMaxColumnLength(), SELECT_MAX, true),
+                mergeAverage(first.getAverageColumnLength(), firstRowCount, second.getAverageColumnLength(), secondRowCount),
+                reduce(first.getNullsCount(), second.getNullsCount(), ADD, false),
+                reduce(first.getDistinctValuesCount(), second.getDistinctValuesCount(), SELECT_MAX, false));
```

Comment:
`null` can indicate 2 thins. 

- statistic is missing (was never computed)
- In case of min/max - that the table is empty

If the table is empty NDV is not gonna be null, but 0.